### ISSUE TITLE
[Bug]: Mass renaming normal file names to names with emojis causing data loss

### ISSUE THREAD

**Issue by EDLLT** [Vinzent03/obsidian-git#794]
### Describe the bug

Describe the bug
Apologies for the empty issue
(This is a recent bug that I discovered faced yesterday so I don't have the necessary information yet. 
I will be slowly updating it with information and tests)

### Relevant errors (if available) from notifications or console (`CTRL+SHIFT+I`)

```shell
Relevant errors (if available) from notifications or console (`CTRL+SHIFT+I`)
```

### Steps to reproduce

Steps to reproduce

### Expected Behavior

Expected Behavior

### Addition context

Addition context

### Operating system

Android

### Installation Method

None

### Plugin version

2.27.0

**Comment by EDLLT** [comment]
```
~/.../Obsidian_TestVault/AI $ ls -la LM\ Studio.md 
-rw-rw----. 1 root everybody 0 Oct 26 16:02 'LM Studio.md'
~/.../Obsidian_TestVault/AI $ file LM\ Studio.md 
LM Studio.md: empty
~/.../Obsidian_TestVault/AI $ cat LM\ Studio.md 
In an ideal case extreme case, the maximum gpu offload layers is 28
26 GPU offload layers for stable usage(higher than that, the model might crash)

## Memory Leak
When you fail to load an LLM model within LM Studio, it still takes up vram. This in return causes you to not be able to fully load the model within the GPU.
The solution is to close LM Studio and run it successfully the first time whilst checking `nvidia-smi`

## Cuda vs Vulkan
You can load up your models in LM Studio using either Cuda or vulkan. Since we have an nvidia GPU, using Cuda is preferable and faster(we need to make sure we have [[Nvidia#^108bb3|Cuda installed]])
```

In here, this is the termux output(similar results when using adb)
My guess is that this is a namespace issue as I had faced very similar issues in the past when I had root.

The interesting thing is that this does not occur to all files, but rather, it occurs randomly(and the larger the commit history, the more likely it'd happen)

TL;DR: The android system sees the file as empty(and expectedly, git), but using cat proves otherwise.

**Comment by EDLLT** [comment]
Okay, after thinking about it for a while, perhaps it's being caused by multiple writes to the same file when checking out from a large commit history size to the current file. Maybe there are multiple writes that are happening to the same file which causes a race condition somehow?

**Comment by EDLLT** [comment]
@Vinzent03 Hey, I am not sure whether if this is considered to be an Obsidian plugin issue or if it's an Obsidian issue with their Adapter and Vault APIs.
Should I open an issue in Obsidian's forums instead?
